File: app/src/main/java/es/etologic/mahjongscoring2/domain/use_cases/CreateGameUseCase.java
package es.etologic.mahjongscoring2.domain.use_cases;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

import es.etologic.mahjongscoring2.data.repositories.GamesRepository;
import es.etologic.mahjongscoring2.domain.model.Game;
import es.etologic.mahjongscoring2.domain.model.GameWithRounds;
import io.reactivex.Single;

public class CreateGameUseCase {

    private GamesRepository gamesRepository;

    @Inject
    public CreateGameUseCase(GamesRepository gamesRepository) { this.gamesRepository = gamesRepository; }

    public Single<GameWithRounds> createGame() {
        List<String> playersNames = new ArrayList<>(4);
        playersNames.add("Player 1");
        playersNames.add("Player 2");
        playersNames.add("Player 3");
        playersNames.add("Player 4");
        Game newGame = new Game(playersNames);
        return gamesRepository.insertOne(newGame)
                .flatMap(gamesRepository::getOneWithRounds);
    }
}

File: app/src/main/java/es/etologic/mahjongscoring2/domain/use_cases/CreateGameUseCase.kt
package es.etologic.mahjongscoring2.domain.use_cases

import es.etologic.mahjongscoring2.data.repositories.GamesRepository
import es.etologic.mahjongscoring2.domain.model.Game
import es.etologic.mahjongscoring2.domain.model.GameWithRounds
import io.reactivex.Single
import java.util.*
import javax.inject.Inject

class CreateGameUseCase @Inject
constructor(private val gamesRepository: GamesRepository) {
    
    internal fun createGame(): Single<Long> {
        val playersNames = ArrayList<String>(4)
        playersNames.add("Player 1")
        playersNames.add("Player 2")
        playersNames.add("Player 3")
        playersNames.add("Player 4")
        val newGame = Game(playersNames)
        return gamesRepository.insertOne(newGame)
    }
}
